code/game/machinery/firealarm.dm

```diff
@@ -326,3 +326,109 @@
+	max_integrity = 100
+	armor = list("melee" = 50, "bullet" = 20, "laser" = 20, "energy" = 20, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 30)
+	var/obj/item/reagent_containers/glass/beaker/internal
+	var/refill_loaded = FALSE
+	req_access = list(ACCESS_ATMOSPHERICS)
+	var/cooldown = 0
+
+/obj/machinery/sprinkler/Initialize()
+	. = ..()
+	internal = new /obj/item/reagent_containers/glass/beaker(src)
+
+/obj/machinery/sprinkler/interact(mob/user)
+	if(!allowed(user))
+		to_chat(user, "<span class='danger'>Access denied.</span>")
+		return
+	dispense()
+
+/obj/machinery/sprinkler/fire_act()
+	.=..()
+	if(refill_loaded)
+		dispense()
+
+/obj/machinery/sprinkler/proc/dispense()
+	if(cooldown < world.time && refill_loaded)
+		var/datum/effect_system/smoke_spread/chem/smoke_machine/smoke = new()
+		smoke.set_up(internal.reagents, 5, 8,  get_turf(src))
+		smoke.start()
+		cooldown = world.time + 200
+		if(!internal.reagents)
+			refill_loaded = FALSE
+			update_icon()
+
+/obj/machinery/sprinkler/update_icon()
+	. = ..()
+	if(refill_loaded)
+		icon_state = "sprinkler1"
+		return
+	icon_state = "sprinkler0"
+
+/obj/item/sprinkler_refill
+	name = "sprinkler refill (water)"
+	desc = "Apply this refill if the sprinkler light is red."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler_refill_water"
+	item_state = "fire_extinguisher"
+	w_class = WEIGHT_CLASS_SMALL
+	var/obj/item/reagent_containers/glass/beaker/refill
+	var/used = FALSE
+	var/chemtype = /datum/reagent/water
+
+/obj/item/sprinkler_refill/Initialize()
+	. = ..()
+	refill = new /obj/item/reagent_containers/glass/beaker(src)
+	refill.reagents.add_reagent(chemtype, 50)
+
+/obj/item/sprinkler_refill/examine(mob/user)
+	. = ..()
+	if(used)
+		. += "It has been used up, better trow it in the trash."
+
+/obj/item/sprinkler_refill/update_icon()
+	. = ..()
+	cut_overlays()
+	if(used)
+		add_overlay("refill_empty")
+		return
+	add_overlay("refill_[max(round(refill.reagents.total_volume/10),1)]")
+
+/obj/item/sprinkler_refill/attack_obj(obj/O, mob/living/user)
+	if(!used && istype(O, /obj/machinery/sprinkler))
+		var/obj/machinery/sprinkler/S = O
+		refill.reagents.trans_to(S.internal, 50, transfered_by = user)
+		playsound(src.loc, 'sound/effects/refill.ogg', 50, 1, -6)
+		to_chat(user, "<span class='notice'>You refill the [S]</span>")
+		S.refill_loaded = TRUE
+		S.update_icon()
+		if(!refill.reagents.total_volume)
+			used = TRUE
+		update_icon()
+
+/obj/item/sprinkler_refill/foam
+	name = "sprinkler refill (firefighting foam)"
+	icon_state = "sprinkler_refill_foam"
+	chemtype = /datum/reagent/firefighting_foam
+
+/obj/item/deployable_sprinkler
+	name = "deployable sprinkler"
+	desc = "A self deploying sprinkler, just press the botton to deploy it."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "sprinkler_d"
+
+/obj/item/deployable_sprinkler/attack_self(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>You start the activating procedure of the [src]...</span>")

```

Comment:
`activating` > `activation`. Or, alternatively, take @nemvar's suggestion and go for `deploying`.